CondCore/SiPixelPlugins/plugins/SiPixelDynamicInefficiency_PayloadInspector.cc

```diff
@@ -731,6 +733,168 @@ namespace {
+  public:
+    SiPixelDynamicInefficiencyPUParametrization()
+        : PlotImage<SiPixelDynamicInefficiency, SINGLE_IOV>("SiPixelDynamicInefficiency Map") {
+      label_ = "SiPixelDynamicInefficiencyPUParameterization";
+      payloadString =
+          fmt::sprintf("%s Dynamic Inefficiency parametrization", SiPixDynIneff::factorString[SiPixDynIneff::pu]);
+    }
+
+    bool fill() override {
+      gStyle->SetPalette(1);
+      auto tag = PlotBase::getTag<0>();
+      auto iov = tag.iovs.front();
+      std::shared_ptr<SiPixelDynamicInefficiency> payload = this->fetchPayload(std::get<1>(iov));
+
+      if (payload.get()) {
+        SiPixDynIneff::PUFactorMap theMap = payload->getPUFactors();
+        std::vector<uint32_t> detIdmasks_db = payload->getDetIdmasks();
+
+        if (!SiPixDynIneff::checkPhase(SiPixelPI::phase::one, detIdmasks_db)) {
+          edm::LogError(label_) << label_ << " maps are not supported for non-Phase1 Pixel geometries !";
+          TCanvas canvas("Canv", "Canv", 1200, 1000);
+          SiPixelPI::displayNotSupported(canvas, 0);
+          std::string fileName(m_imageFileName);
+          canvas.SaveAs(fileName.c_str());
+          return false;
+        }
+
+        std::vector<std::vector<double> > listOfParametrizations;
+
+        // retrieve the list of phase1 detids
+        const auto& reader =
+            SiPixelDetInfoFileReader(edm::FileInPath(SiPixelDetInfoFileReader::kPh1DefaultFile).fullPath());
+        const auto& p1detIds = reader.getAllDetIds();
+
+        // fill the maps
+        for (const auto& det : p1detIds) {
+          const auto& values = SiPixDynIneff::getMatchingPUFactors(det, theMap, detIdmasks_db);
+          if (!std::count(listOfParametrizations.begin(), listOfParametrizations.end(), values)) {
+            listOfParametrizations.push_back(values);
+          }
+        }
+
+        unsigned int depth = listOfParametrizations.size();
+
+        // functional for polynomial of n-th degree
+        auto func = [](double* x, double* p) {
+          int n = p[0];
+          double* params = p + 1;
+          ROOT::Math::Polynomial pol(n);
+          return pol(x, params);
+        };
+
+        std::vector<TF1*> parametrizations;
+        std::vector<std::string> formulas;
+        parametrizations.reserve(depth);
+
+        int index{0};
+        for (auto& params : listOfParametrizations) {
+          //for(auto& [detid,params] : theMap) {
+          index++;
+          double xmin = 0.;
+          double xmax = 25.;
+          int n = params.size();
+          int npar = n + 2;
+          TF1* f1 = new TF1((fmt::sprintf("region: %i", index)).c_str(), func, xmin, xmax, npar);
+          params.insert(params.begin(), n);
+          double* arr = params.data();
+          f1->SetLineWidth(2);
+          f1->SetParameters(arr);
+          parametrizations.push_back(f1);
+
+          // debug
```

Comment:
Are these debug lines supposed to stay there?

Reply:
I suppose we lost already the next IB, right? It would have been useful to have this integrated to help with https://its.cern.ch/jira/browse/PDMVRELVALS-188 ASAP.

Reply:
... which is not the answer to my question.
Trying to reformulate: were these debug lines just forgotten after the development of the code, or are they expected to remain there forever? If they must be removed, please do so now, relaunch the tests, and ask @francescobrivio to re-sign right after: we have plenty of time before 2300 IB

Reply:
Ah, ok, you fixed it already: thank you @mmusich

Reply:
> were these debug lines just forgotten after the development of the code, or are they expected to remain there forever? 

they can be expected to remain forever. I changed them from `cout` to logMessages anyway.

> we have plenty of time before 2300 IB

well... I am not sure.